carve.lic

```diff
@@ -0,0 +1,118 @@
+=begin
+  Suggestions and contributions are welcome: https://github.com/rpherbig/dr-scripts
+=end
+
+custom_require(%w(common common-crafting))
+
+class Carve
+  include DRC
+  include DRCC
+
+  def initialize(args:[])
+    settings = get_settings
+    @bag = settings.crafting_container('backpack')
+    @finish, @type, @chapter, @recipe_name, @material, @noun = args
+
+    case @type
+    when /carving/
+      carve_item
+    else
+      echo('Carve does not yet support that form of carving.')
+    end
+  end
+
+  def get_or_fail(item)
+    waitrt?
+    case bput("get my #{item}", '^You get', '^You are already', '^You pick up', '^What do you', '^What were you')
+    when 'What do you', 'What were you'
+      echo("You seem to be missing: #{item}")
+      exit
+    end
+  end
+
+  def store_thing(item)
+    waitrt?
+    bput("put my #{item} in my #{@bag}", 'You put your', 'What were you referring to')
+  end
+
+  def turn_to(section)
+    bput("turn my book to #{section}", 'You turn your', 'The book is already')
+  end
+
+  def carve_item
+    get_or_fail('carving book')
+    turn_to("chapter #{@chapter}")
+    turn_to("page #{find_recipe(@chapter, @recipe_name)}")
+    bput('study my book', 'Roundtime')
+    store_thing('book')
+    get_or_fail('chisel')
+    get_or_fail("#{@material} rock")
+    carve('cut my rock with my chisel')
+  end
+
+  def carve(command)
+    waitrt?
+    case bput(command,
+              'rough, jagged edges',
+              'determine it is no longer level',
+              'developed an uneven texture along its surface',
+              #'Applying the final touches',
+              'You cannot figure out how to do that.',
+              'you see some discolored areas',
+              'Roundtime'
+             )
+    when 'rough, jagged edges'
+      waitrt?
+      store_thing(checkright)
+      get_or_fail('rifflers')
+      command = "rub my #{@noun} with my rifflers"
+    when 'determine it is no longer level', 'developed an uneven texture along its surface'
+      waitrt?
+      store_thing(checkright)
+      get_or_fail('rasp')
+      command = "rub my #{@noun} with my rasp"
+    when 'you see some discolored areas'
+      waitrt?
+      store_thing(checkright)
+      get_or_fail('polish')
+      command = "apply my polish to my #{@noun}"
+    #when 'Applying the final touches'
+    when 'You cannot figure out how to do that.'
+      finish
+    else
+      waitrt?
+      store_thing(checkright) unless checkright == "chisels"
+      get_or_fail('chisels') unless checkright == "chisels"
+      command = "cut my #{@noun} with my chisels"
+    end
+    waitrt?
+    carve(command)
+  end
+
+  def finish
+    waitrt?
+    store_thing(checkright)
+    case @finish
+    when /log/
+      logbook_item
+    when /stow/
+      store_thing(@noun)
+    when /trash/
+      dispose_trash(@noun)
+    end
+    # NOTE: using 2 volume recipes leaves no leftover material when ordering a small rock
```

Comment:
The note should say 3 volume recipes.  I initially commented the lines out with the intent that carve would be used with recipes that only use a small rock(3 volumes), meaning no leftover material.  When you order the rock, they give you a deed as boulders are too heavy to pick up.  Some reworking of the script could be done to stow the deed and get/tap it once you're in a workroom.  If recipes requiring other volumes are used, cleanup will be required.